server/src/main/java/org/elasticsearch/common/settings/SettingsUpdater.java

```diff
@@ -28,10 +29,10 @@
 public final class SettingsUpdater {
     final Settings.Builder transientUpdates = Settings.builder();
     final Settings.Builder persistentUpdates = Settings.builder();
-    private final ClusterSettings clusterSettings;
+    private final AbstractScopedSettings scopedSettings;
 
-    public SettingsUpdater(ClusterSettings clusterSettings) {
-        this.clusterSettings = clusterSettings;
+    public SettingsUpdater(AbstractScopedSettings scopedSettings) {
```

Comment:
Does this class make sense for `IndexScopedSettings`? If not, maybe we should have separate explicit constructors for `ClusterSettings` and `ProjectScopedSettings` instead?